run_vllm_forward.py

```diff
@@ -0,0 +1,110 @@
+###############################################################################
+# Copyright (C) 2024 Habana Labs, Ltd. an Intel Company
+###############################################################################
+
+import argparse
+import torch
+from vllm import LLM, SamplingParams
+from vllm.sequence import SequenceData, SequenceGroupMetadata, ExecuteModelRequest
+
+def setup_profiler():
+    activities = [torch.profiler.ProfilerActivity.CPU]
+    activities.extend([torch.profiler.ProfilerActivity.HPU])
+    wait = 0
+    active = 1
+    warmup = args.steps - active
+
+    schedule = torch.profiler.schedule(wait=wait, warmup=warmup, active=active, repeat=1)
+    profiler = torch.profiler.profile(
+        schedule=schedule,
+        activities=activities,
+        on_trace_ready=torch.profiler.tensorboard_trace_handler('.', use_gzip=True),
+        record_shapes=False,
+        with_stack=True)
+    return profiler
+
+def kill_process():
+    """Kills python3 main process manually"""
+    print("Killing process manually")
+    import psutil
+    for proc in psutil.process_iter():
+        if proc.name() == "python3":
```

Comment:
That's a bit brutal... Could we just kill 'ray' instances? Besides this can lead to a race condition when this script gets killed before ray.

Reply:
what about wrapping the job in separate process that we spawn and forcibly killing that process afterwards?

Reply:
something like this: https://github.com/vllm-project/vllm/issues/1908#issuecomment-2141297771

Reply:
Tried solution from this issue, works better :)